core/ledger/kvledger/snapshot.go

```diff
@@ -114,12 +118,19 @@ func (l *kvLedger) generateSnapshotMetadataFiles(
 	if err != nil {
 		return err
 	}
+
+	stateDBType := l.config.StateDBConfig.StateDatabase
+	if stateDBType != ledger.CouchDB {
+		stateDBType = simpleKeyValueDB
+	}
```

Comment:
- We could have been explicit, if we fix upfront when we parse the configs, to make sure that these two are the choices. Unfortunately, in the current code, the [check](https://github.com/hyperledger/fabric/blob/7fd80f54fd9fe6ee5f7460558bf884cbb2ba2996/core/ledger/kvledger/txmgmt/privacyenabledstate/db.go#L67) is like this only. Which means that any value (practically, default value) other than "CouchDB" for this string would result in leveldb as a statedb.
- Yes, it's nothing to do with the data format of export. The difference it important to make sure that we do not end up loading leveldb data into couchdb (which is more restrictive)
- As mentioned in point 1 above, exact value could very well be empty string on one peer while "goleveldb" being specified on the other (theoretically, it could be anything)

Reply:
I see. Make sense. Because of the existing mechanism, this looks fine.